### ISSUE TITLE
Firestick 4k max (files with hdr10+ and dolby vision profile 8.1)

### ISSUE THREAD

**Issue by bobaoapae** [jellyfin/jellyfin-androidtv#2630]
### Describe the bug

When playing a file that has hdr10+ and also dolby vision profile 8.1 image get full black or duplicate([check here](https://preview.redd.it/vt2p4ee0c2oa1.jpeg?width=4032&format=pjpg&auto=webp&v=enabled&s=606052d50a1bf4922cf1951cc532a63c737beeca)).

This issue occours on all others players too, but kodi appers to have found a solution ([here](https://forum.kodi.tv/showthread.php?tid=371557)). Whe using this version i go do options -> video -> uncheck hdr10+ and video play smoth using Dolby Vision, if keep hdr10+ and dolby vision same issue happens, appers to have some problem with firestick decode that can't play a file with theses two codecs enabled, need to explicit configure player to use one or other

I can play the same file using this kodi modded version without any issue, but using exoplayer/libvlc on jellyfind don't work

### Media info of the file

```shell
Vídeo
Título: 4K HEVC HDR
Codec: HEVC
Perfil: Main 10
Nível: 150
Resolução: 3832x1596
Proporção da tela: 2.40:1
Entrelaçado: No
Taxa de quadros: 23.976025
Bitrate: 25167 kbps
Profundidade de bits: 10 bit
Intervalo do vídeo: HDR
Video range type: HDR10
DV title: DV Profile 8.1 (HDR10)
DV version major: 1
DV version minor: 0
DV profile: 8
DV level: 6
DV rpu preset flag: 1
DV el preset flag: 0
DV bl preset flag: 1
DV bl signal compatibility id: 1
Espaço de cor: bt2020nc
Transferência de cor: smpte2084
Cores primárias: bt2020
Formato de pixel: yuv420p10le
Quadros de referência: 1
```

### Application version

latest

### Device information

Amazon Firestick 4k max

### Android version

latest

### Jellyfin server version

latest

**Comment by bobaoapae** [comment]
Issue also reported on ExoPlayer now 

https://github.com/google/ExoPlayer/issues/11096

**Comment by SubEdict** [comment]
@bobaoapae  Thank you very much, will test as soon as I can.

**Comment by bobaoapae** [comment]
I hope this issue receive some attention.. this it's a old bug already know on plex community but never get fixed by major players, kodi it's the first to release some kind of fix.

Libvlc has the same issue, mxplayer, justplayer, exoplayer.. and many others i tested, much of theses to depend on libvlc/exoplayer.

So if exoplayer or libvlc receive some fix this can propagate for several others players.

I already tested some kinds of "fix" in the code by my self but i really don't know nothing about codecs it's self to build the misssig logic arround... in my opnion player should check if has hdr10+ and dolby vision send just the best layer supported by display, at least on firestick due to have some bug on the decoder in the system that cause this bug.

**Comment by SubEdict** [comment]
Honesty, I have no idea how this issue doesn't get the highest exposure on the web. Where are the original devs? How come they didn't release this Kodi fork fix in the original nightly versions.

BTW, I see that the latest release was 3+ month ago (Jan 24), is it still maintained? There were so many other bug fixes in the original Kodi git. Any suggestion where else can I look for more updated/sync version?

Also, I already link your post to anyone else as the no'1 explanation + fix. :)

**Comment by KyleSanderson** [comment]
> Honesty, I have no idea how this issue doesn't get the highest exposure on the web. Where are the original devs? How come they didn't release this Kodi fork fix in the original nightly versions.
> 
> BTW, I see that the latest release was 3+ month ago (Jan 24), is it still maintained? There were so many other bug fixes in the original Kodi git. Any suggestion where else can I look for more updated/sync version?
> 
> Also, I already link your post to anyone else as the no'1 explanation + fix. :)

The vast majority of jellyfin users are transcoding 480i and thinking that's the best they're going to get. Kodi is a bit better at 720p (down to 240p), but leaning on the latest codec and metadata combinations takes work from a developer who just isn't watching IPTV (which, unfortunately, is the vast majority).

Yes, it's a pretty pathetic situation honestly. However, there's a couple people trying to actually improve the ecosystem despite constant egos and "this worked on dialup", "media bitrates never exceed 1mbit/s", "Linux doesn't support HDR so I don't care", and every other fallacy under the sun.

**Comment by frezeen** [comment]
+1  need fix

**Comment by mikemulhearn** [comment]
I have this exact same issue with Fire TV 4K Max with Plex (same view as OP’s screenshot with the picture wrapping). I have no issues playing the same video through Plex on a 4K Fire TV (TV w/ Fire TV built-in). Wondering if this issue is specific to Fire TV 4K Max?

**Comment by rbjtech-github** [comment]
Yes it is - issue is the FireTV 4K Max.  I have got around the issue by converting/fixing the DV8.1/HDR or DV8.1/HDR10+ media files - they now play just fine with Emby for FireTV (have yet to try JF - I'm just looking to transition..).    Files prior to conversion worked fine on FireTV 4K, so issue is the Max I agree.

**Comment by tonihei** [comment]
I got around to investigate this a bit with a FireTV Stick 4K Ultra HD. It has the same decoders available I believe and it also fails to play the provided file with the Dolby Vision decoder on the device (despite the decoder claiming it supports this profile and level). If I force the HEVC fallback level decoder, it works just fine (but is of course no longer using Dolby Vision). 

@ybai001 : Not sure if you also maintain these codecs, but it seems the Amazon Fire TV sticks have 3 different Dolby Vision codecs, and one of them has matching profile/level information to play the file in this issue: `OMX.MTK.VIDEO.DECODER.DVHE.STH`. However, the output is just black frames. Given the surprising number of reports in this issue recently, could it be that this was caused by some system update with a new codec binary?

If this issue is more permanent and can't be fixed by Dolby/Amazon, we could add a workaround in ExoPlayer to exclude this codec on these devices, but ideally we can just fix the underlying issue. 

@ all issue reporters: The Kodi "fix" is basically what I described here: exclude the Dolby Vision decoder and force the fallback HEVC level decoder to play the video. You can do this manually in your ExoPlayer-based app by adding something like this on `DefaultRenderersFactory` (which can be set in `ExoPlayer.Builder`):
```
.setMediaCodecSelector(
  (mimeType, requiresSecureDecoder, requiresTunnelingDecoder) -> {
    if (mimeType.equals(MimeTypes.VIDEO_DOLBY_VISION)) {
      return ImmutableList.of();
    }
    return MediaCodecSelector.DEFAULT.getDecoderInfos(
      mimeType, requiresSecureDecoder, requiresTunnelingDecoder);
  })
``

**Comment by quietvoid** [comment]
> @ all issue reporters: The Kodi "fix" is basically what I described here: exclude the Dolby Vision decoder and force the fallback HEVC level decoder to play the video. 

No, that is not the case.
You tested on the old 4K stick, not the 2021 4K Max.

On the 4K Max, even when forcing the regular HEVC decoder, it is still playing in Dolby Vision and outputs a blank image.

The only workaround has been to strip the Dolby Vision or HDR10+ from bitstream **before** passing the bytes to the decoder.

**Comment by rbjtech-github** [comment]
Are you sure the issue is not poorly created DV8 files ?   I have seen the black output myself, but by running the 'correction/remux' using DoviTools - the files then play fine (with DV being triggered).  Same with creating DV8 files from DV5 + HDR10, they play fine as well.   Using the latest Exo in Emby actually, but assume JF issue would be the same.

**Comment by tonihei** [comment]
> On the 4K Max, even when forcing the regular HEVC decoder, it is still playing in Dolby Vision and outputs a blank image.
Only the 4K Max behaves that way, it completely ignores the chosen decoder.

This sounds very unusual. Are you certain it's not using the Dolby Vision decoder? You can verify this by looking at the logs in a bugreport or logcat. 

> Are you sure the issue is not poorly created DV8 files ?

I was working on the assumption that these a perfectly valid files that are also supported by other players. If the files are just broken media, then there is probably not much that can be done in the player and the media should be fixed instead.

**Comment by quietvoid** [comment]
> This sounds very unusual. Are you certain it's not using the Dolby Vision decoder? You can verify this by looking at the logs in a bugreport or logcat.

Yes, the issue even affects users with displays that don't support Dolby Vision.

> Are you sure the issue is not poorly created DV8 files ? I have seen the black output myself, but by running the 'correction/remux' using DoviTools - the files then play fine (with DV being triggered). 

The files are perfectly valid here, it's just not super common for both HDR10+ and Dolby Vision to be in the same bitstream.
I have no idea what this "correction" is doing but maybe it's just simply removing HDR10+.

**Comment by rbjtech-github** [comment]
I've had issues with DV8 (DV/HDR10) files with the same black screen symptom on the Max - a run though the correction/remux fixes those - so I assumed (incorrectly) that it fixed the DV/HDR10+ files as well.   I just tested two 'fixed' DV/HDR10+ files and I still get the black screen.  The standard DV8/HDR10 files are now all ok.  So please disregard the above about it being fixed - it's not.

Apologies - my mistake !

**Comment by bobaoapae** [comment]
The test files run without problem in the same version of jellyfin on the TV it's self, issue only in the firestick.

Also this happens in all the majors players i tested on the firestick, it's a bug in the firestick but amazon don't assume the problem.

The only working solution it's the solution provided by some fork of Kodi, to enable disable Dolby or HDR10+, doing exactly what @quietvoid said, they remove the bytes from the stream.

**Comment by ybai001** [comment]
> I got around to investigate this a bit with a FireTV Stick 4K Ultra HD. It has the same decoders available I believe and it also fails to play the provided file with the Dolby Vision decoder on the device (despite the decoder claiming it supports this profile and level). If I force the HEVC fallback level decoder, it works just fine (but is of course no longer using Dolby Vision).
> 
> @ybai001 : Not sure if you also maintain these codecs, but it seems the Amazon Fire TV sticks have 3 different Dolby Vision codecs, and one of them has matching profile/level information to play the file in this issue: `OMX.MTK.VIDEO.DECODER.DVHE.STH`. However, the output is just black frames. Given the surprising number of reports in this issue recently, could it be that this was caused by some system update with a new codec binary?
> 
> If this issue is more permanent and can't be fixed by Dolby/Amazon, we could add a workaround in ExoPlayer to exclude this codec on these devices, but ideally we can just fix the underlying issue.
> 
> @ all issue reporters: The Kodi "fix" is basically what I described here: exclude the Dolby Vision decoder and force the fallback HEVC level decoder to play the video. You can do this manually in your ExoPlayer-based app by adding something like this on `DefaultRenderersFactory` (which can be set in `ExoPlayer.Builder`):
> 
> ```
> .setMediaCodecSelector(
>   (mimeType, requiresSecureDecoder, requiresTunnelingDecoder) -> {
>     if (mimeType.equals(MimeTypes.VIDEO_DOLBY_VISION)) {
>       return ImmutableList.of();
>     }
>     return MediaCodecSelector.DEFAULT.getDecoderInfos(
>       mimeType, requiresSecureDecoder, requiresTunnelingDecoder);
>   })
> ``
> ```

@tonihei , I'll check with Dolby internal stakeholder about this issue. Stay tuned.

BR,
Yanning

**Comment by ybai001** [comment]
@tonihei , our Dolby QA tested some Dolby Vision Profile 8, Level = 6, CCID = 1 contents on Amazon Firestick 4K max. There is no this kind of issue. Could you help to share the media with me for testing? You can send a link to ybai@dolby.com.

**Comment by bobaoapae** [comment]
> @tonihei , our Dolby QA tested some Dolby Vision Profile 8, Level = 6, CCID = 1 contents on Amazon Firestick 4K max. There is no this kind of issue. Could you help to share the media with me for testing? You can send a link to [ybai@dolby.com](mailto:ybai@dolby.com).

Can I send? I have some files here with this problem in firestick 4k max.

**Comment by ybai001** [comment]
> > @tonihei , our Dolby QA tested some Dolby Vision Profile 8, Level = 6, CCID = 1 contents on Amazon Firestick 4K max. There is no this kind of issue. Could you help to share the media with me for testing? You can send a link to [ybai@dolby.com](mailto:ybai@dolby.com).
> 
> Can I send? I have some files here with this problem in firestick 4k max.

@bobaoapae, tonihei has already sent us the test stream and we can reproduce this issue now. The corresponding engineers are analyzing this issue. Stay tuned.

**Comment by theinfguy** [comment]
Is there an update on this?

Also having major issues playing back content on the 4K Max.

**Comment by ybai001** [comment]
Let me check with team whether they can provide some feedback early this week. Thanks for your patient.

**Comment by ybai001** [comment]
Here is feedback from corresponding team.

First of all, they did some testing. Playing the stream with HDR10+ and Dolby Vision stream on Firestick and Amazon Cube TV
- FireStick 4k Max – Video playback is black, Audio is Good.
- Amazon Fire TV Cube – Video playback is fine (selects Dolby Vision), Audio is Good.
Exoplayer used is v2.11.apk from Dolby Vision Source SDK v2.6 LA.

So, as of now it looks like on few devices having good playback and other devices it is black.

The team is still consulting the Amazon team why the variation is observed on amazon devices.

**Comment by bobaoapae** [comment]
This problem it's a little old... just got more attention theses days.

reference: https://forum.makemkv.com/forum/viewtopic.php?t=29757

**Comment by Marvellll** [comment]
Im having the same issue of black screen on dv hdr files on my firestick 4k max. I hope there will be a fix.  I can provide files with the issue if needed

**Comment by bobaoapae** [comment]
@ybai001 hello any news on that?

**Comment by ybai001** [comment]
> @ybai001 hello any news on that?

Our engineering team is consulting Amazon team but has not got feedback by now. :(

**Comment by NDDDDDDDDD** [comment]
damn not even dolby getting a response from anazon? they should really give out refunds for their shitty device

**Comment by bobaoapae** [comment]
> > @ybai001 hello any news on that?
> 
> Our engineering team is consulting Amazon team but has not got feedback by now. :(

I don't think Amazon will fix this, it's a old bug as you can see here [https://forum.makemkv.com/forum/viewtopic.php?t=29757](https://forum.makemkv.com/forum/viewtopic.php?t=29757). This will need to be solved in https://github.com/google/ExoPlayer or more exactly in the new https://github.com/androidx/media

**Comment by quietvoid** [comment]
For info the new FireTV Stick 4K Max 2nd gen (2023) still has the same issues.

Dolby Vision profile 7 FEL still shows a blank screen.
And files with both HDR10+ and Dolby Vision also show a blank screen.

**Comment by mellmo** [comment]
just to confirm, the Amazon Fire TV Cube should be able to play 4k hdr10+ just fine right ? if not can someone recommend something that works ?

**Comment by SubEdict** [comment]
> For info the new FireTV Stick 4K Max 2nd gen (2023) still has the same issues.

Just about to buy. Thanks for saving me the time and headache of trying to install the regular version.

**Comment by vyngur** [comment]
From my limited experience, this seems to happen with files with with an aspect ratio different from 16:9.
I have files with the same HDR10/DV8.1 profiles working as expected on a 4K Max Fire TV if they're 16:9, but in 2.40:1 I get a black screen (no playback interface) with just audio.

**Comment by quietvoid** [comment]
> From my limited experience, this seems to happen with files with with an aspect ratio different from 16:9.

No, that's unrelated. You might want to check that your non working files also have HDR10+.

**Comment by vyngur** [comment]
This works:
```
Video
Title: 4K HEVC HDR
Codec: HEVC 
Profile: Main 10
Level: 150
Resolution: 3840x2160
Aspect ratio: 16:9
Interlaced: No
Framerate: 23.976025
Bitrate: 25117 kbps
Bit depth: 10 bit
Video range: HDR
Video range type: HDR10
DV title: DV Profile 8.1 (HDR10)
DV version major: 1
DV version minor: 0
DV profile: 8
DV level: 6
DV rpu preset flag: 1
DV el preset flag: 0
DV bl preset flag: 1
DV bl signal compatibility id: 1
Color space: bt2020nc
Color transfer: smpte2084
Color primaries: bt2020
Pixel format: yuv420p10le
Ref frames: 1
```
This doesn't
```
Video
Title: 4K HEVC HDR
Codec: HEVC
Profile: Main 10
Level: 150
Resolution: 3840x1608
Aspect ratio: 2.40:1
Interlaced: No
Framerate: 23.976025
Bitrate: 24689 kbps
Bit depth: 10 bit
Video range: HDR
Video range type: HDR10
DV title: DV Profile 8.1 (HDR10)
DV version major: 1
DV version minor: 0
DV profile: 8
DV level: 6
DV rpu preset flag: 1
DV el preset flag: 0
DV bl preset flag: 1
DV bl signal compatibility id: 1
Color space: bt2020nc
Color transfer: smpte2084
Color primaries: bt2020
Pixel format: yuv420p10le
Ref frames: 1
```

**Comment by quietvoid** [comment]
Jellyfin doesn't show HDR10+. Use MediaInfo instead.

**Comment by tonihei** [comment]
> This will need to be solved in https://github.com/google/ExoPlayer or more exactly in the new https://github.com/androidx/media

ExoPlayer is generally happy to accept workarounds for known device-specific issues. The problem to my understanding is that the only known workaround is extremely verbose and involves modifying the actual bitstream. It would be a lot easier if the workaround code is more confined and manageable, e.g. selecting specific decoder instances, providing additional config flags to these decoders etc. If anyone can find a way to solve the playback problem in this way, please let us know.

**Comment by bobaoapae** [comment]
> > This will need to be solved in https://github.com/google/ExoPlayer or more exactly in the new https://github.com/androidx/media
> 
> ExoPlayer is generally happy to accept workarounds for known device-specific issues. The problem to my understanding is that the only known workaround is extremely verbose and involves modifying the actual bitstream. It would be a lot easier if the workaround code is more confined and manageable, e.g. selecting specific decoder instances, providing additional config flags to these decoders etc. If anyone can find a way to solve the playback problem in this way, please let us know.

Yeah... but the only solution that works it's the one provided by @quietvoid in they fork of Kodi, as the bug it's in the driver embbed by amazon in the firestick that refuse to handle the bitstream correclty if hdr10+ and dolby vision are provided togheter. So yes would be much more better if anyone found a better solution, but has be a long time of this bug in theses devices without any action, if possible to use this currenlty "fix" for this problem in this way until someone can figure out a better way would be amazing for us with this problem. As currently the only way it's using the fork of Kodi or reenconding the file.

**Comment by quietvoid** [comment]
>  It would be a lot easier if the workaround code is more confined and manageable, e.g. selecting specific decoder instances, providing additional config flags to these decoders etc. If anyone can find a way to solve the playback problem in this way, please let us know.

@tonihei If you actually have suggestions of configurations to test, then it could be possible to try other workarounds.

I don't know much about Android. The only thing I could test was disallowing the Dolby Vision codecs. 
That didn't work on these devices.
Other workarounds don't seem too likely, or it just needs someone who actually knows what can be tried.

**Comment by MichaelRUSF** [comment]
@NetixQc  I have the Fire Stick 4k (2nd Gen, 2023) non max version and this bug also affects it.

**Comment by bobaoapae** [comment]
@ybai001 just checking again, any news?

**Comment by ybai001** [comment]
> @ybai001 just checking again, any news?

@bobaoapae, let me check with relative engineer whether we got feedback from Amazon.

**Comment by AndreaFontanaDynamisPRC** [comment]
Had the same issue - Firestick 4K 2023. Black screen and audio playing. Solved the problem by unchecking both "Audio Bitstream Dolby Digital" and "Bitstream Audio DTS". Honestly even if my hardware is incompatible with these settings, I don't understand why they should cause the black screen issue.

**Comment by daf0x** [comment]
@ybai001

I am writing to express my frustration with the lack of communication and resolution regarding this issue. It has been months since it was first reported, and we have yet to receive any updates or feedback from you. This is unacceptable and unprofessional.

We understand that issues can arise, but the lack of transparency and accountability is frustrating. We deserve better than this. We have been waiting for months without any resolution or communication from you. This is not acceptable.

We urge you to take responsibility for the problem and communicate with us in a timely and transparent manner. We expect you to provide us with a clear timeline for when the issue will be resolved. We have been patient for too long, and we need a resolution as soon as possible.

We suggest that you provide regular updates on the progress of the issue and communicate with us in a transparent and timely manner. We hope that you will take our concerns seriously and work towards a resolution as soon as possible.

Thank you for your attention to this matter.

**Comment by JPVenson** [comment]
> @ybai001
> 
> I am writing to express my frustration with the lack of communication and resolution regarding this issue. It has been months since it was first reported, and we have yet to receive any updates or feedback from you. This is unacceptable and unprofessional.
> 
> We understand that issues can arise, but the lack of transparency and accountability is frustrating. We deserve better than this. We have been waiting for months without any resolution or communication from you. This is not acceptable.
> 
> We urge you to take responsibility for the problem and communicate with us in a timely and transparent manner. We expect you to provide us with a clear timeline for when the issue will be resolved. We have been patient for too long, and we need a resolution as soon as possible.
> 
> We suggest that you provide regular updates on the progress of the issue and communicate with us in a transparent and timely manner. We hope that you will take our concerns seriously and work towards a resolution as soon as possible.
> 
> Thank you for your attention to this matter.

Dear user.
I am very sory that you have this kind of unpleasant opinion of an app that you paid absolutely nothing for and did not contribute anything apart from this complaint. 

I would recommend you to read though the whole discussion here. But i will be so nice to give you a summery: 
"The cause of the issue was at first not easy to identify and after that was determined to be an underlying issue was an issue with amazon's OS"

I will now take my responsibility and communicate to you that there is no current progress made on this issue. If you want it fixed however you are welcome to do so and PR the changes if you dont want to wait on anyone to fix it in their spare time for you to enjoy.

I tried to formulate this response as Professional as possible so please be lenient if it may sounded a bit harsh.

**Comment by ybai001** [comment]
> @ybai001
> 
> I am writing to express my frustration with the lack of communication and resolution regarding this issue. It has been months since it was first reported, and we have yet to receive any updates or feedback from you. This is unacceptable and unprofessional.
> 
> We understand that issues can arise, but the lack of transparency and accountability is frustrating. We deserve better than this. We have been waiting for months without any resolution or communication from you. This is not acceptable.
> 
> We urge you to take responsibility for the problem and communicate with us in a timely and transparent manner. We expect you to provide us with a clear timeline for when the issue will be resolved. We have been patient for too long, and we need a resolution as soon as possible.
> 
> We suggest that you provide regular updates on the progress of the issue and communicate with us in a transparent and timely manner. We hope that you will take our concerns seriously and work towards a resolution as soon as possible.
> 
> Thank you for your attention to this matter.

@daf0x , sorry for making you frustrated. As I replied in Aug, 21 in this thread, our Dolby engineering team has analyzed this issue. We observed the playback is OK on some Amazon devices while it is not on other Amazon devices. We provided same Dolby codec release to Amazon and we are helping to consult the Amazon team why the variation is observed. Dolby team has checked with Amazon team several times. The last time is on Dec 15th. Unfortunately, we didn't get any feedback by now.

**Comment by KyleSanderson** [comment]
> > @ybai001
> > I am writing to express my frustration with the lack of communication and resolution regarding this issue. It has been months since it was first reported, and we have yet to receive any updates or feedback from you. This is unacceptable and unprofessional.
> > We understand that issues can arise, but the lack of transparency and accountability is frustrating. We deserve better than this. We have been waiting for months without any resolution or communication from you. This is not acceptable.
> > We urge you to take responsibility for the problem and communicate with us in a timely and transparent manner. We expect you to provide us with a clear timeline for when the issue will be resolved. We have been patient for too long, and we need a resolution as soon as possible.
> > We suggest that you provide regular updates on the progress of the issue and communicate with us in a transparent and timely manner. We hope that you will take our concerns seriously and work towards a resolution as soon as possible.
> > Thank you for your attention to this matter.
> 
> @daf0x , sorry for making you frustrated. As I replied in Aug, 21 in this thread, our Dolby engineering team has analyzed this issue. We observed the playback is OK on some Amazon devices while it is not on other Amazon devices. We provided same Dolby codec release to Amazon and we are helping to consult the Amazon team why the variation is observed. Dolby team has checked with Amazon team several times. The last time is on Dec 15th. Unfortunately, we didn't get any feedback by now.

Yeah it's not on you guys at all, it's a common industry complaint with that team. These sticks are to deliver Prime Video, if Prime works nothing else matters (as unfortunate as it is).

As discovered in another issue (and I hadn't checked the repo history in a long time until then) the project is maintained now by a single individual. All the information that's needed is here, if you need DoVi use any other player until the new backend comes in.

**Comment by nreis96** [comment]
Hi, I have created this issue for exoPlayer https://github.com/androidx/media/issues/957 which is currently one of the native players in Stremio and both firestick 4k max 1st and 2nd gen will cause issues (either black screen with audio only or split screen).

Considering a player as Vimu Media Player makes it work I was hoping a fix can be created within the player itself (exoPlayer in this case), if anyone has suggestions or comments please add them in there or here!

Thanks in advance.

**Comment by nreis96** [comment]
@tonihei "If this issue is more permanent and can't be fixed by Dolby/Amazon, we could add a workaround in ExoPlayer to exclude this codec on these devices, but ideally we can just fix the underlying issue." Considering it's been half a year and this quite an annoying issue, could you please add a workaround in ExoPlayer itself? Even if not preferred solution? It's quite obvious by now that Amazon won't do anything, and DV team (@ybai001) has tried their best. Let users know please

**Comment by tonihei** [comment]
The issue that the workaround is not quite as simple as I thought. What I was seeing only fixes the issue on older FireTV Stick 4K Ultra HD devices, not the 2021 4K Max ones. The only known workaround for the new devices (as far as I understand) is to modify the actual bitstream to remove either HDR+ or DV metadata (see https://github.com/quietvoid/xbmc/commit/308135d42e04eca6c2e50af9d510a4b1c0daae4e). While this is possible in theory, it's quite invasive as bitstream rewriting isn't usually done when processing video bytes and there is no existing method we could simply hook into. If you are motivated to solve it, feel free to send a Media3 pull request that does that and we can see how it looks like. It could probably go somewhere in [this method](https://github.com/androidx/media/blob/59afb4fb0121035d0a071c42b947f81d9ddd1548/libraries/exoplayer/src/main/java/androidx/media3/exoplayer/video/MediaCodecVideoRenderer.java#L1157) where you could try to modify the contents of `inputBuffer.data`.

**Comment by nreis96** [comment]
There's no chance to edit it the way Kodi has it in advanced config settings to force HDR or DV?

**Comment by massijay** [comment]
I think one solution might be that the client (with a certain setting) asks to the server to strip out DoVi or HDR10+ metadata on the fly. In this way we can entirely workaround this problem

**Comment by GreenDude120** [comment]
4k max user here; i can confirm that kodi's advanced option they added at some point in 21 omega solved the issue for me. I would love jellyfin to add one of the suggestions above whereby an advanced setting could make the server strip out the metadata on the fly. please & thanks!

**Comment by nickshanks347** [comment]
Same here for Jellyfin on a 4k Max (just bought so guessing 2023 model)

**Comment by mfizz1** [comment]
I too have the same issue. But whilst the team are working hard to fix it, is there a script or way to remove hdr10+ elements from the file so that only dolby vision remains?

**Comment by mfizz1** [comment]
For anyone interested, by running the following command, you can get rid of the HDR10+ data which then means the file is playable again, for hevc files.

`ffmpeg -i input.mkv -c copy -bsf:v "filter_units=remove_types=39" output.mkv`

**Comment by mfizz1** [comment]
Does anyone know of a way that avoids remux using ffmpeg?

**Comment by Skyl3lazer** [comment]
I'm still interested in this fix as well! It's frustrating that it's a bug on their end but a workaround would be nice.

**Comment by nickshanks347** [comment]
> I'm still interested in this fix as well! It's frustrating that it's a bug on their end but a workaround would be nice.

Workaround is using Kodi to strip either HDR10 or DV metadata on the fly

**Comment by Skyl3lazer** [comment]
That workaround would involve my non-technical users having to install
third party media players and plugins, it's a nonstarter for me
unfortunately.

**Comment by dandud100** [comment]
I have the same issue

**Comment by Tresillo2017** [comment]
> This also affects the new Google TV Streamer which uses the same SoC as Firestick 4k Max

Can confirm, I can play movies with the new av1 codec but other ones with hvenc and hdr10+ with dolby vision and atmos it will give a playback error or hang the app on a black screen without audio

**Comment by mfizz1** [comment]
2 years since the original issue was reported. This seems to be a bug on all Android systems where they are connected to a TV, arguably one of the most used ways of consuming content, yet there seems to be no traction on this issue. Kodi have worked out a way to filter the hdr10 or dolby vision profile. Can we not do the same.

**Comment by thornbill** [comment]
> Can we not do the same.

Sure! Feel free to open a PR to do so.

**Comment by twinko** [comment]
> > For anyone interested, by running the following command, you can get rid of the HDR10+ data which then means the file is playable again, for hevc files.
> > `ffmpeg -i input.mkv -c copy -bsf:v "filter_units=remove_types=39" output.mkv`
> 
> Thanks, it worked. You should add `-map 0` otherwise all secondary audio and subtitle tracks will be deleted! `ffmpeg -i input.mkv -map 0 -c copy -bsf:v "filter_units=remove_types=39" output.mkv`

HI @dfriday 
Where to put this code? I guess ich have to modifiy output/input.mkv right?

**Comment by bobaoapae** [comment]
Yes this issue it's still present

**Comment by daf0x** [comment]
> This issue has gone 120 days without comment. To avoid abandoned issues, it will be closed in 21 days if there are no new comments.
> 
> If you're the original submitter of this issue, please comment confirming if this issue still affects you in the latest release or master branch, or close the issue if it has been fixed. If you're another user also affected by this bug, please comment confirming so. Either action will remove the stale label.
> 
> This bot exists to prevent issues from becoming stale and forgotten. Jellyfin is always moving forward, and bugs are often fixed as side effects of other changes. We therefore ask that bug report authors remain vigilant about their issues to ensure they are closed if fixed, or re-confirmed - perhaps with fresh logs or reproduction examples - regularly. If you have any questions you can reach us on [Matrix or Social Media](https://docs.jellyfin.org/general/getting-help.html).

yes, the issue is still here. No, there’s still no communication.

**Comment by metaljay** [comment]
Tbf I’m completely shocked there still isn’t a bug fix from Amazon for this

**Comment by GreenDude120** [comment]
> Tbf I’m completely shocked there still isn’t a bug fix from Amazon for this

im not. it doesnt affect their paid services like Prime Video and their latest devices have it fixed so they can sell more. yay profits

**Comment by dandud100** [comment]
https://github.com/androidx/media/issues/1895

**Comment by metaljay** [comment]
> > Tbf I’m completely shocked there still isn’t a bug fix from Amazon for this
> 
> im not. it doesnt affect their paid services like Prime Video and their latest devices have it fixed so they can sell more. yay profits

I’m using the latest Amazon firestick, (4k gen 2) and I still have this bug

**Comment by GreenDude120** [comment]
> I’m using the latest Amazon firestick, (4k gen 2) and I still have this bug

im talking about the latest gen cube. firestick 4k gen2 is not new, already a couple years old

**Comment by metaljay** [comment]
> > I’m using the latest Amazon firestick, (4k gen 2) and I still have this bug
> 
> im talking about the latest gen cube. firestick 4k gen2 is not new, already a couple years old

The latest cube is the 3rd Gen and the firestick is the 4k max gen 2, these have the same hardware and came out the same year. Am I missing something?

**Comment by crobibero** [comment]
Opening as the fix still requires changes to AndroidTV

**Comment by mfizz1** [comment]
From what I understand, this issue is unlikely to ever be fixed due to a limitation in **ExoPlayer**—it doesn’t have the ability to distinguish or filter HDR elements. The HDR processing is handled entirely by **MediaCodec** (Android’s decoder), and there’s no known way to interact with it to adjust HDR behavior.  

As a workaround, I’ve been using **Jellyfin through Kodi** with the **Jellyfin plugin**, which works well since Kodi doesn’t rely on ExoPlayer.  

The only real fix would require **Amazon to update their ExoPlayer implementation** with a software patch. The latest implementation has better automatic HDR handling.

**Comment by Pr0mises** [comment]
@mfizz1 which plugin are you using? Jellyfin or jellycon

**Comment by mfizz1** [comment]
> @mfizz1 which plugin are you using? Jellyfin or jellycon

Jellyfin, with native paths.

**Comment by SonOfAbaddon** [comment]
As I am also affected with my FireTV4Kmax v1:

Would it be possible to have a toggle in the Jellyfin client that, if enabled, tells the server to strip either DV or HDR/HDR+ and transcode?

**Comment by mfizz1** [comment]
> As I am also affected with my FireTV4Kmax v1:
> 
> Would it be possible to have a toggle in the Jellyfin client that, if enabled, tells the server to strip either DV or HDR/HDR+ and transcode?

I think you'd be better off putting an automated file manager to strip away hdr10+ or dv data. They are quite easy to set up using basic templates. Do it once and it's applied to all files, rather than transcoding every time.

**Comment by mfizz1** [comment]
Well, it does "remotely" solve it for those with either hdr10+ or dv so at least some can play it. For those interested you can utilize something like tdarr or unmanic.

In practice, this is difficult to overcome given that some tvs report back support for all formats and have their own internal fallback. So unless you have a database of such tv support what and implement this logic or have a better method for identifying. Once you work it out,  please submit a pull request

**Comment by nielsvanvelzen** [comment]
Closing as duplicate of #4021
